plugin-copilot/src/main/java/appland/copilotChat/NavieCopilotChatRequestHandler.java

```diff
@@ -230,11 +234,16 @@ public void end() {
         chatSession.ask(listener,
                 copilotModel,
                 asCopilotChatMessages(openAIRequest.messages()),
+                collectProxiedRequestHeaders(fullHttpRequest),
                 openAIRequest.temperature(),
                 openAIRequest.topP(),
                 openAIRequest.n());
     }
 
+    private @NotNull Map<String, String> collectProxiedRequestHeaders(@NotNull FullHttpRequest fullHttpRequest) {
+        return fullHttpRequest.headers().entries().stream().collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
Does it make sense to filter the headers? I'm slightly concerned about leaking transactional headers like `content-type`, `accept`, `content-size`, `if-updated-since` etc.